### ISSUE TITLE
<w3m-button> does not reconnect on page reload [bug] 

### ISSUE THREAD

**Issue by layinka** [WalletConnect/web3modal#2334]
### Link to minimal reproducible example

https://github.com/layinka/Angular-starter-test

### Summary

I am using **Wagmi** and **Web3Modal** in an **Angular 17 project**. I use the **w3m-button** web component
Before recent versions of WalletModal, whenever i refresh the page, the w3m-button detects if my dapp was previously connected and updates the status of the button to show the connected wallet, now anytime i refresh the page, the w3m-button does not seem to be able to detect if i was connected before

Secondly, if i initiate a manual reconnection by calling the **reconnect** wagmi function, wagmi will detect that i am connected but the w3m-button still does not recognize that i am connected

### List of related npm package versions

"@wagmi/connectors": "^5.0.6",
"@wagmi/core": "^2.10.4",
"@web3modal/wagmi": "^4.2.2",
"viem": "^2.12.5",

**Comment by layinka** [comment]
Yes,I noticed the rendering issues too, was hoping you will notice as it
didn't happen Everytime

**Comment by glitch-txs** [comment]
Please provide a minimal example of the implementation. Closing for now.

**Comment by layinka** [comment]
A Minimal example was included in my initial post.
I have added the link below again

https://github.com/layinka/Angular-starter-test

**Comment by glitch-txs** [comment]
Please make it minimal 🙏

**Comment by layinka** [comment]
It is minimal,

It's a minimal angular project

Not sure there is any other way to minimize it

It only shows connect button

**Comment by glitch-txs** [comment]
why do you have react and ethers as a dependency? Could you make it minimal from scratch please?

**Comment by layinka** [comment]
I have **React** as a dependency because both **@walletconnect/modal-core** , **valtio** and use-sync-external-store ( all used by either viem, wagmi, walletconnect or wallet modal) has a dependency on react and without react the app wont work, 

Without adding **react** , i get the error

`[ERROR] Could not resolve "react"`

I added ethers because of the wagmi ethers adapter, but i can easily remove that
